### PR TITLE
base.yaml: Add shard_thief_password and commented out stealing towns.

### PR DESCRIPTION
**Pull request by Sarvatt**
just the towns currently in base-stealing.yaml

### CODE REVIEW

File: data/base-stealing.yaml [hunk 1]
```diff
@@ -5926,11 +5926,4 @@
   pawnable: false
   room: 13721
   town: Horse Clan
-- exceptionally_max: 305
```
Comment (rpherbig) [inline data/base-stealing.yaml]: Why was this removed?

Comment (Sarvatt) [reply data/base-stealing.yaml]: It is in a citizen only shop in shard and shouldn't really go in there, sorry for lumping it together with the other change
